crates/esthri/src/tempfile.rs

```diff
@@ -21,6 +22,9 @@ pub const TEMP_FILE_PREFIX: &str = ".esthri_temp";
 impl TempFile {
     pub async fn new(dir: PathBuf, suffix: Option<&str>) -> Result<Self> {
         let suffix = suffix.unwrap_or_default().to_owned();
+        if !&dir.exists() {
+            fs::create_dir_all(&dir)?;
+        }
```

Comment:
should be 

```rust
tokio::fs::create_dir_all(&dir).await?;
```

in an async context you want to stick to async versions of functions if they exist. a blocking call inside an async runtime can gum up the executor which can manifest as a performance hit or deadlock. for something like this it's not a huge deal because I doubt that function blocks for long but its an easy fix.

and I think you should be able to drop the `&` in `!&dir.exists()` - it should just return a bool

Reply:
the reason for this is that `async` in rust implements [cooperative multitasking](https://en.wikipedia.org/wiki/Cooperative_multitasking) - so it's up to author to make sure nothing takes too much cpu time before yielding to other tasks